tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py

```diff
@@ -0,0 +1,372 @@
+            ),
+            pytest.param(
+                "v2", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-5410")
+            ),
+        ]
+
+
+@green_squad
+@pytest.mark.parametrize(
+    argnames=argnames,
+    argvalues=argvalues,
+)
+@tier2
+@skipif_ocs_version("<4.8")
+@kms_config_required
+@skipif_managed_service
+@skipif_hci_provider_and_client
+@skipif_disconnected_cluster
+@skipif_proxy_cluster
+class TestPvcExpansionWhenFull(ManageTest):
+    """
+    Tests to verify encrypted PVC expansion when the PVC is 100% utilized.
+    Verify utilization alert will stop firing after volume expansion.
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        kv_version,
+        kms_provider,
+        pv_encryption_kms_setup_factory,
+        use_vault_namespace,
+        project_factory,
+        storageclass_factory,
+        create_pvcs_and_pods,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.kms = pv_encryption_kms_setup_factory(kv_version, use_vault_namespace)
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create a project
+        self.proj_obj = project_factory()
+
+        # Create an encryption enabled storageclass for RBD
+        self.sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.kms.kmsid,
+        )
+
+        if kms_provider == constants.VAULT_KMS_PROVIDER:
+            # Create ceph-csi-kms-token in the tenant namespace
+            self.kms.vault_path_token = self.kms.generate_vault_token()
+            self.kms.create_vault_csi_kms_token(namespace=self.proj_obj.namespace)
+
+        self.pvc_size = 4
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=self.pvc_size,
+            access_modes_rbd=[constants.ACCESS_MODE_RWO],
+            access_modes_cephfs=[constants.ACCESS_MODE_RWO],
+            sc_rbd=self.sc_obj,
+            project=self.proj_obj,
+        )
+
+    def test_pvc_expansion_when_full(self, threading_lock):
+        """
+        Verify PVC expansion when the encrypted PVC is 100% utilized.
+
+        Steps:
+            1. Create a encrypted storageclass.
+            2. Create a encrypted PVC of size 4G using defined vault configuration.
+            3. create a pod and mount the PVC inside the pod.
+            4. Fill FS 100% using FIO workload.
+            5. Expand PVC to size 10G
+            6. continue IO workload and verify the IOs are succeeding
+            7. Verify utilization alert will stop firing after volume expansion.
+
+        """
+        pvc_size_expanded = 10
+
+        # Run IO to utilise 100% of volume
+        log.info("Run IO on all to utilise 100% of PVCs")
+        for pod_obj in self.pods:
```

Comment:
If you'd like you can make them run parallel and save some time. Here is an example [https://github.com/paraggit/ocs-ci/blob/377ff1d814ba702eb1d870570f1ada12198cf654/tests/lvmo/test_lvm_multi_clone.py#L136-L162](https://github.com/paraggit/ocs-ci/blob/377ff1d814ba702eb1d870570f1ada12198cf654/tests/lvmo/test_lvm_multi_clone.py#L136-L162)